pkgs/by-name/ld/ld64/package.nix

```diff
@@ -0,0 +1,184 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchurl,
+  darwin,
+  libtapi,
+  libunwind,
+  llvm,
+  llvmPackages,
+  meson,
+  ninja,
+  openssl,
+  pkg-config,
+  python3,
+  swiftPackages,
+  xar,
+  gitUpdater,
+}:
+
+let
+  # The targetPrefix is prepended to binary names to allow multiple binuntils on the PATH to both be usable.
+  targetPrefix = lib.optionalString (
+    stdenv.targetPlatform != stdenv.hostPlatform
+  ) "${stdenv.targetPlatform.config}-";
+
+  # ld64 needs CrashReporterClient.h, which is hard to find, but WebKit2 has it.
+  # Fetch it directly because the Darwin stdenv bootstrap can’t depend on fetchgit.
+  crashreporter_h = fetchurl {
+    url = "https://raw.githubusercontent.com/apple-oss-distributions/WebKit2/WebKit2-7605.1.33.0.2/Platform/spi/Cocoa/CrashReporterClientSPI.h";
+    hash = "sha256-0ybVcwHuGEdThv0PPjYQc3SW0YVOyrM3/L9zG/l1Vtk=";
+  };
+  # First version with all the required definitions
```

Comment:
I wanted to avoid depending on xnu, particularly for private headers. It also makes things easier for Linux if it doesn’t have to deal with Darwin SDKs if possible.

Reply:
I also expanded the comment here.